.github/GITHUB_WORKFLOW.md

```diff
@@ -0,0 +1,299 @@
+  - [7. (If Needed) Keep Your Branch in Sync](#7-if-needed-keep-your-branch-in-sync)
+  - [8. Create a Pull Request](#8-create-a-pull-request)
+  - [9. Get a Code Review](#9-get-a-code-review)
+  - [10. Squash Commits](#10-squash-commits)
+  - [11. Merging a commit](#11-merging-a-commit)
+  - [(If Needed) Reverting a Commit](#if-needed-reverting-a-commit)
+
+## 1. Fork in the Cloud
+
+1. Visit the repo in which would like to contribute to
+2. Click `Fork` button (top right) to establish a cloud-based fork.
+
+## 2. Clone Fork to Local Storage
+
+Set `user` to match your github profile name:
+
+```bash
+export user=<your github profile name>
+```
+
+Both `$working_dir` and `$user` are mentioned in the figure above.
+
+Create your clone:
+
+```bash
+mkdir -p $working_dir
+cd $working_dir
+git clone git@github.com:$user/deepgram-go-sdk.git
+
+cd $working_dir/deepgram-go-sdk
+git remote add upstream git@github.com:deepgram/deepgram-go-sdk.git
+
+# Never push to upstream main (or master)
+git remote set-url --push upstream no_push
+
+# Confirm that your remotes make sense:
+git remote -v
+```
+
+## 3. (Optional But HIGHLY Recommended) Set `git pull` to `rebase` Instead
+
+This is an optional step and we will do our best here to provide directions should you not set this, but doing a `git pull` is invasive and can get quite nasty. The preferred way is to **ALWAYS** use **rebase** in favor of a traditional `git pull`. 
+
+This will override `git pull` to effectively alias or under the covers do a `git rebase` globally:
+
+```bash
+git config --global pull.rebase true
+```
+
+If you only want to do this per repo, you can navigate to the root of the repo and run this:
+
+```bash
+git config pull.rebase true
+```
+
+## 4. Create a Working Branch
+
+Get your local `main` up to date. Note that depending on which repository you are working from, the default branch may be called `master` instead of `main`.
+
+```bash
+cd $working_dir/deepgram-go-sdk
+git fetch upstream
+git checkout main
+git rebase upstream main
+```
+
+Create your new branch.
+
+```bash
+git checkout -b myfeature
+```
+
+You may now edit files on the `myfeature` branch.
+
+## 5. Commit Your Changes
+
+You will probably want to regularly commit your changes. It is likely that you will go back and edit, build, and test multiple times. After a few cycles of this, you might [amend your previous commit](https://www.w3schools.com/git/git_amend.asp).
+
+```bash
+git commit
+```
+
+## 6. Push to GitHub
+
+When your changes are ready for review, push your working branch to your fork on GitHub.
+
+```bash
+git push --set-upstream origin myfeature
+```
+
+If got an error with this 
+
+## 7. (If Needed) Keep Your Branch in Sync
+
+You will need to periodically fetch changes from the `upstream` repository to keep your working branch in sync. Note that depending on which repository you are working from, the default branch may be called `master` instead of `main`.
```

Comment:
Do we need to say this? 

> Note that depending on which repository you are working from, the default branch may be called `master` instead of `main`.